### ISSUE TITLE
Ok awesome I get the first issue :-(. Network scanner never returning, begin never returning

### ISSUE THREAD

**Issue by rchancey** [telit/arduino-tlt-library#2]
@fabiopi-tlt 
I purchased the Charlie after getting stuck with the MKR 1500 and their firmware issues.. got this board today and very excited. I installed the telit library from here : https://github.com/telit/arduino-charlie and got the board recognized.
I am able to do the blink and fade no problem. Those work.

I am working to send an SMS. I have a valid SIM card and my APN is mobilenet. (I was able to send from the MKR 1500 but not receive.. which is what I need).
I updated my APN in the sample and pushed it.. and it gets stuck on the **TLTAccess.begin().** I attach my entire sketch below.

Please help.. anything I can do to debug this? since I am the first I think whatever we do for me will help others. There are other folks in the same situation looking at this Charlie board.

I am located in Texas in the US and using SpeedTalk SMS card also tried my Hologram SIM card

I get the same 'stuck' where it never gets past **TLTAccess.begin()**

I really want to get this working on the Charlie board. Please help!

Here is my sketch and the Serial output:
NB IoT/LTE Cat M1 networks scanner
Connecting...

`
Sketch:
/*Copyright (C) 2020 Telit Communications S.p.A. Italy - All Rights Reserved.*/
/*    See LICENSE file in the project root for full license information.     */

/**
  @file
    - TLTMDM.h

  @brief
    Scan Networks.

  @details
     This example sketch prints the IMEI number of the modem, then check if it is connected to an operator.\n
     It then scans nearby networks and prints their signal strength.
  @version 
    1.0.0
  
  @note

  @author
    Cristina Desogus

  @date
    09/23/2021
 */
// libraries

// initialize the library instance
#include "TLTMDM.h"

ME310* myME310 = new ME310();

TLT TLTAccess(myME310);  // include a 'true' parameter to enable debugging
TLTScanner scannerNetworks(myME310);

// Save data variables
String IMEI = "";

// serial monitor result messages
String errortext = "ERROR";

char APN[] = "mobilenet";

void setup() {
  // initialize serial communications and wait for port to open:
  Serial.begin(115200);
  myME310->begin(115200);
  delay(2000);
  Serial.println("NB IoT/LTE Cat M1 networks scanner");
  scannerNetworks.begin();

  // connection state
  boolean connected = false;

  // Start module
  // If your SIM has PIN, pass it as a parameter of begin() in quotes
  Serial.println("Connecting...");
  while (!connected) {
    if (TLTAccess.begin(NULL, APN, true) == READY) {
      connected = true;
    } else {
      Serial.println("Not connected");
      delay(1000);
    }
  }

  // get modem parameters
  // IMEI, modem unique identifier
  Serial.print("Modem IMEI: ");
  IMEI = myME310->request_imei_software_version();
  IMEI.replace("\n", "");
  if (IMEI != NULL) {
    Serial.println(IMEI);
  }
}

void loop() {
  // currently connected carrier
  Serial.print("Current carrier: ");
  Serial.println(scannerNetworks.getCurrentCarrier());

  // returns strength and BER
  // signal strength in 0-31 scale. 31 means power > 51dBm
  // BER is the Bit Error Rate. 0-7 scale. 99=not detectable
  Serial.print("Signal Strength: ");
  Serial.print(scannerNetworks.getSignalStrength());
  Serial.println(" [0-31]");

  // scan for existing networks, displays a list of networks
  Serial.println("Scanning available networks. May take some seconds.");
  Serial.println(scannerNetworks.readNetworks());
  // wait ten seconds before scanning again
  delay(10000);
}
`

**Comment by fabiopi-tlt** [comment]
Hello,
Since the charlie modem might be trying to register in NBIoT (which may take a lot of time), how much did you wait in the TLTAccess.begin() call?

things you can do to try and debug:

Serial.println("Connecting...");
**TLTAccess.setTimeout(20000);** /*set a 20 seconds timeout*/

while (!connected) {
if (TLTAccess.begin(NULL, APN, true, **false**) == READY) {  /*disable the syncronous flag*/
connected = true;
} else {
Serial.println("Not connected");
delay(1000);
}
}

also, before line https://github.com/telit/arduino-tlt-library/blob/main/src/TLT.cpp#L226
add a 
Serial.println(_readyState); 

to check in which state the library is.

We will add additional debug prints in the library in a future release.

In any case, if you plan to send an SMS, you should look also at the example sketches:

https://github.com/telit/arduino-tlt-library/tree/main/examples/SendSMS_example
https://github.com/telit/arduino-tlt-library/tree/main/examples/ReceiveSMS_example

**Comment by rchancey** [comment]
Ok... I am using the sketch you pasted.. all I did was add my APN. I tried what you suggested and getting not connected. Here is a screen shot:
<img width="663" alt="Screen Shot 2021-10-12 at 9 47 54 AM" src="https://user-images.githubusercontent.com/29101431/136978770-62258f3e-85fa-4dbc-b102-c95eac96c10c.png">
:

Is there a way we can check to ensure the board is actually working? I added the prinln to the cpp but nothing printed.

**Comment by fabiopi-tlt** [comment]
hello, can you keep pressed the on_off button on the board for 3-4 seconds? My feeling is that the modem is not on.

**Comment by fabiopi-tlt** [comment]
> @fabiopi-tlt is there a basic sketch I can run, and others, that does a simple self test of the board? I am sure these were tested but this simple use case should work right out of box. I have a feeling perhaps a component is bad. Perhaps a set of commands?

you can use the examples directly from ME310 library:

https://github.com/telit/arduino-me310-library/tree/main/examples/M2M_example

or https://github.com/telit/arduino-me310-library/blob/main/examples/ME310_AT_Test/ME310_AT_Test.ino

**Comment by fabiopi-tlt** [comment]
ok, the modem is OK. If you run the first sketch now, what output do you get? Please consider that if you are using a NBIoT SIM card, the registration can take a while. Have you plugged the LTE antenna?

**Comment by Humancell** [comment]
Yeah, I'm done for today.  No matter what I do I can't get the green blinky light working any more.  I had that one partial success, and now nothing no matter what I do.  I look forward to some feedback from the Telit team.  So close.

**Comment by fabiopi-tlt** [comment]
hello, we pushed a new version of the library, it allows to print additional debug info if the TLTAccess constructor is called as in the new[ ScanNetworks example](https://github.com/telit/arduino-tlt-library/blob/main/examples/ScanNetworks_example/ScanNetworks_example.ino#L32):

TLT TLTAccess(myME310, true); 

Also, all the examples now automatically turn the ME310 on at the beginning of the code.
